crates/db-device/README.md

```diff
@@ -0,0 +1,109 @@
+# db-device
+
+The db-device crate mainly contains:
+
+- `DeviceIo` and `DeviceIoMut` trait: defining the read/write operation on specific device
+- `IoManager`: managing all devices of virtual machine, (un)register their io resource callback
+- `IoManagerContext` trait: transaction context for `IoManager` to (un)register devices
+- `ResourceConstraint`, `Resource` and `DeviceResources`: abstractions for requiring and describing resources of a device
+
+# Design
+
+The db-device crate is designed to support the virtual machine's device model.
+
+The core concepts of device model are [Port I/O](https://wiki.osdev.org/I/O_Ports) and [Memory-mapped I/O](https://en.wikipedia.org/wiki/Memory-mapped_I/O), which are two main methods of performing I/O between CPU and devices.
+
+The `DeviceIo` and `DeviceIoMut` trait are used to define the operations of devices in vm when guest OS performing I/O to emulated devices.
+
+In addition, we need to let CPU know the I/O operation would be send to which device, so the devices also need to declare the resources they used for I/O operation, which are defined in `Resource` enum. It also defines the other resources that the device will use, such as interrupts, KVM slots and MAC addresses. And `DeviceResources` struct gethers a series resources of a device together.
+
+When a device is creating, it may not yet know the specific parameters of the resources it uses, but only the constraint of resources it needs, so `ResourceConstraint` enum is used to describe the resources required of each device.
```

Comment:
describe the resources required of each device. -> describe the resources requirement for each device.

that could make this sentence clearer

Reply:
Thanks, changed.